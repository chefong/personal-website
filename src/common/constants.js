import hamburgerMenuAnimation from '~/assets/menu-animation.json';

export const gaEvents = {
    PAGE_RENDERED: 'page_rendered',
    ABOUT_NAV_CLICKED: 'about_nav_clicked',
    PROJECTS_NAV_CLICKED: 'projects_nav_clicked',
    EXPERIENCE_NAV_CLICKED: 'experience_nav_clicked',
    THEME_TOGGLE_CLICKED: 'theme_toggle_clicked',
    RESUME_CLICKED: 'resume_clicked',
    CONTACT_ME_CLICKED: 'contact_me_clicked',
    GITHUB_CLICKED: 'github_icon_clicked',
    LINKEDIN_CLICKED: 'linkedin_icon_clicked',
    SPOTIFY_CLICKED: 'spotify_icon_clicked',
    MEDIUM_CLICKED: 'medium_icon_clicked',
    MAIL_CLICKED: 'mail_icon_clicked',
};

export const navItems = [
    {
        to: 'about',
        num: '01.',
        name: 'About',
        clickEventId: gaEvents.ABOUT_NAV_CLICKED,
    },
    {
        to: 'projects',
        num: '02.',
        name: 'Projects',
        clickEventId: gaEvents.PROJECTS_NAV_CLICKED,
    },
    {
        to: 'experience',
        num: '03.',
        name: 'Experience',
        clickEventId: gaEvents.EXPERIENCE_NAV_CLICKED,
    },
];

export const projects = [
    {
        nameId: 'PROJECTS_PATCH',
        descriptionId: 'PROJECTS_PATCH_DESCRIPTION',
        githubLink: 'https://github.com/willshiao/Patch',
    },
    {
        nameId: 'PROJECTS_CHEATERBEATER',
        descriptionId: 'PROJECTS_CHEATERBEATER_DESCRIPTION',
        githubLink: 'https://github.com/willshiao/CheaterBeater',
    },
    {
        nameId: 'PROJECTS_BAUT',
        descriptionId: 'PROJECTS_BAUT_DESCRIPTION',
        githubLink: 'https://github.com/KirbyDownB/GEICOChatBot',
    },
    {
        nameId: 'PROJECTS_NUTSHELL',
        descriptionId: 'PROJECTS_NUTSHELL_DESCRIPTION',
        githubLink: 'https://github.com/chefong/nutshell',
    },
    {
        nameId: 'PROJECTS_STOCK_OVERFLOW',
        descriptionId: 'PROJECTS_STOCK_OVERFLOW_DESCRIPTION',
        githubLink: 'https://github.com/kimjihwan0208/stockoverflow',
    },
    {
        nameId: 'PROJECTS_JOYRIDE',
        descriptionId: 'PROJECTS_JOYRIDE_DESCRIPTION',
        githubLink: 'https://github.com/ericong18/joyride',
    },
];

export const experiences = [
    {
        imagePath: {
            light: 'imgs/experience/opentable.png',
            dark: 'imgs/experience/opentable-white.png',
        },
        nameId: 'EXPERIENCE_OPENTABLE',
        positionId: 'SOFTWARE_ENGINEER_1',
        duration: 'February 2021 - Present',
        teamId: 'EXPERIENCE_OPENTABLE_TEAM_2',
        initiativesId: 'EXPERIENCE_OPENTABLE_INITIATIVES_2',
        technologiesId: 'EXPERIENCE_OPENTABLE_TECHNOLOGIES_2',
    },
    {
        imagePath: {
            light: '/imgs/experience/intuit.svg',
            dark: '/imgs/experience/intuit-white.svg',
        },
        nameId: 'EXPERIENCE_INTUIT',
        positionId: 'SOFTWARE_ENGINEER_1',
        duration: 'July 2020 - February 2021',
        teamId: 'EXPERIENCE_INTUIT_TEAM',
        initiativesId: 'EXPERIENCE_INTUIT_INITIATIVES',
        technologiesId: 'EXPERIENCE_INTUIT_TECHNOLOGIES',
    },
    {
        imagePath: {
            light: 'imgs/experience/opentable.png',
            dark: 'imgs/experience/opentable-white.png',
        },
        nameId: 'EXPERIENCE_OPENTABLE',
        positionId: 'SOFTWARE_ENGINEERING_INTERN',
        duration: 'June 2019 - August 2019',
        teamId: 'EXPERIENCE_OPENTABLE_TEAM_1',
        initiativesId: 'EXPERIENCE_OPENTABLE_INITIATIVES_1',
        technologiesId: 'EXPERIENCE_OPENTABLE_TECHNOLOGIES_1',
    },
];

export const handles = [
    {
        name: 'GitHub',
        icon: ['fab', 'github'],
        link: 'https://github.com/chefong',
        clickEventId: gaEvents.GITHUB_CLICKED,
    },
    {
        name: 'LinkedIn',
        icon: ['fab', 'linkedin-in'],
        link: 'https://www.linkedin.com/in/ericong18/',
        clickEventId: gaEvents.LINKEDIN_CLICKED,
    },
    {
        name: 'Spotify',
        icon: ['fab', 'spotify'],
        link: 'https://open.spotify.com/user/ericong18?si=H0kHkFdxQ-GyKIlfLWzDqg',
        clickEventId: gaEvents.SPOTIFY_CLICKED,
    },
    {
        name: 'Medium',
        icon: ['fab', 'medium'],
        link: 'https://medium.com/@ericong18',
        clickEventId: gaEvents.MEDIUM_CLICKED,
    },
    {
        name: 'Email',
        icon: 'envelope',
        link: 'mailto:ericong18@gmail.com',
        clickEventId: gaEvents.MAIL_CLICKED,
    },
];

export const likes = [
    'ABOUT_LIKE_TEA',
    'ABOUT_LIKE_WATERMELON',
    'ABOUT_LIKE_MUSIC',
    'ABOUT_LIKE_DESIGN',
    'ABOUT_LIKE_PREMIER_LEAGUE',
];

export const dislikes = ['ABOUT_DISLIKE_CILANTRO'];

export const themeColors = {
    light: '#fffefb',
    dark: '#222222',
    tan: '#f1e7d0',
};

export const lottieProps = {
    loop: false,
    autoplay: false,
    animationData: hamburgerMenuAnimation,
};

export const reactScrollLinkProps = {
    smooth: true,
    offset: -150,
    duration: 750,
};

// Segment frames for menu open and close
export const menuOpenSegments = [30, 70];
export const menuCloseSegments = [100, 140];

// Box shadow styling for when the nav menu is open
export const menuBoxShadow = '0px 4px 28px rgba(0, 0, 0, 0.06)';

// Menu transition styling for menu
export const menuTransition = 'all 0.5s ease';

export const themeOrder = ['light', 'tan', 'dark'];

export const iconSize = '28px';

export const iconColors = {
    light: themeColors.dark,
    dark: themeColors.light,
    tan: themeColors.dark,
};

export const toastOptions = {
    style: {
        padding: '16px',
        fontFamily: 'Greycliff',
        backgroundColor: '#303030',
        color: 'white',
    },
};

export const particlesParams = {
    particles: {
        number: {
            value: 80,
            density: {
                enable: true,
                value_area: 1500,
            },
        },
        line_linked: {
            enable: true,
            opacity: 0.02,
        },
        move: {
            direction: 'right',
            speed: 0.05,
        },
        size: {
            value: 1.25,
        },
        opacity: {
            anim: {
                enable: true,
                speed: 1,
                opacity_min: 0.05,
            },
        },
    },
    retina_detect: true,
};
