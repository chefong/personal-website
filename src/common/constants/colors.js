export default {
    light: '#fffefb',
    dark: '#222222',
    tan: '#f1e7d0',
};
