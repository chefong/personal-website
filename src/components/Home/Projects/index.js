import { FormattedMessage } from 'text-provider';
import styles from './styles.module.scss';
import Card from './Card';
import projects from '~/common/constants/projects';

export default function Projects() {
    return (
        <section
            className={`${styles.section} container-fluid`}
            id="projects"
            data-testid="Projects-container"
        >
            <div className="row justify-content-center">
                <div className="col-lg-9 col-md-9">
                    <div className={styles.headerGroup}>
                        <h2 className={styles.header}>
                            <span className={styles.headerNum}>02. </span>
                            <FormattedMessage id="PROJECTS_HEADING" />
                        </h2>
                    </div>
                </div>
            </div>
            <div className="row justify-content-center">
                <div className="col-lg-9 col-md-9">
                    <div className="row justify-content-center">
                        {projects.map((project) => (
                            <Card {...project} key={project.nameId} />
                        ))}
                    </div>
                </div>
            </div>
        </section>
    );
}
