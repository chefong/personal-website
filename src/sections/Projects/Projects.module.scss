.section {
  margin: 10em auto;
}

.headerGroup {
  margin-bottom: 2em;
}

.header {
  color: var(--theme-page-text);
}

.headerNum {
  color: var(--theme-color);
}