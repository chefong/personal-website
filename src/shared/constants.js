const navItems = [
  {
    to: "about",
    num: "01.",
    name: "About"
  },
  {
    to: "projects",
    num: "02.",
    name: "Projects"
  },
  {
    to: "experience",
    num: "03.",
    name: "Experience"
  }
];

const projects = [
  {
    name: "Patch",
    description: "A web application that retroactively fixes your Zoom audio recordings.",
    githubLink: "https://github.com/willshiao/Patch",
    wonHackathon: true
  },
  {
    name: "CheaterBeater",
    description: "Identify and act on fraudulent Devpost hackathon submissions.",
    githubLink: "https://github.com/willshiao/CheaterBeater",
    wonHackathon: true
  },
  {
    name: "Baut",
    description: "A chatbot that recommends movies and music based on the user's likings.",
    githubLink: "https://github.com/KirbyDownB/GEICOChatBot",
    wonHackathon: true
  },
  {
    name: "birdie",
    description: "A web application that visualizes tweets on a Mapbox map.",
    githubLink: "https://github.com/chefong/birdie",
    wonHackathon: false
  },
  {
    name: "Stock Overflow",
    description: "Use news data to predict how a stock will act in the future.",
    githubLink: "https://github.com/kimjihwan0208/stockoverflow",
    wonHackathon: false
  },
  {
    name: "Joyride",
    description: "Determine the most efficient way to reach a destination with your carpool buddies!",
    githubLink: "https://github.com/ericong18/joyride",
    wonHackathon: true
  }
];

const experiences = [
  {
    imagePath: "/imgs/experience/intuit.png",
    name: "Intuit",
    position: "Software Engineer",
    duration: "July 2020 - Present",
    description: "Financial Data Platform Team."
  },
  {
    imagePath: "imgs/experience/opentable.png",
    name: "OpenTable",
    position: "Software Engineering Intern",
    duration: "June 2019 - August 2019",
    description: "As a part of the Content Team, I worked on various consumer-facing products (internal tools, Restaurant Profile, GuestCenter Admin) using React, TypeScript, Jest, and OpenComponents. One of the projects I worked on focused on improving an existing internal tool to speed up workflow for internal marketing users. Another project I worked on was kickstarting the Gallery Photos feature on OpenTable's GuestCenter Admin with my team, giving restaurants the ability to manage their own photos."
  }
];

const handles = [
  {
    name: "GitHub",
    iconPath: "/imgs/footer/github.svg",
    link: "https://github.com/chefong"
  },
  {
    name: "LinkedIn",
    iconPath: "/imgs/footer/linkedin.svg",
    link: "https://www.linkedin.com/in/ericong18/"
  },
  {
    name: "Instagram",
    iconPath: "/imgs/footer/instagram.svg",
    link: "https://www.instagram.com/chef_ong/"
  },
  {
    name: "Medium",
    iconPath: "/imgs/footer/medium.svg",
    link: "https://medium.com/@ericong18"
  },
  {
    name: "Email",
    iconPath: "/imgs/footer/envelope.svg",
    link: "mailto:ericong18@gmail.com"
  }
];

const revealProps = {
  duration: 500,
  delay: 400,
  distance: "1.25em",
  bottom: true
};

export { navItems, projects, experiences, handles, revealProps };