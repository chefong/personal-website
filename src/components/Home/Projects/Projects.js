import { FormattedMessage } from 'text-provider';
import styles from './Projects.module.scss';
import Card from './Card/Card';
import { projects } from '../../../common/constants';

export default function Projects() {
  return (
    <section className={`${styles.section} container-fluid`} id="projects" data-testid="Projects-container">
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className={styles.headerGroup}>
            <h2 className={styles.header}>
              <span className={styles.headerNum}>02. </span>
              <FormattedMessage id="PROJECTS_HEADING" />
            </h2>
          </div>
        </div>
      </div>
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className="row justify-content-center">
            {projects.map((project) => <Card {...project} key={project.nameId} />)}
          </div>
        </div>
      </div>
    </section>
  );
}
