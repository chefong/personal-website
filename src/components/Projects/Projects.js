import styles from './Projects.module.css';
import Card from './Card/Card';
import { projects } from '../../shared/constants';

export default function Projects() {
  return (
    <section className={`${styles.section} container-fluid`} id="projects">
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className={styles.headerGroup}>
            <h2 className={styles.header}><span className={styles.headerNum}>02.</span> Projects</h2>
          </div>
        </div>
      </div>
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className="row justify-content-center">
            {projects.map(project => <Card { ...project } />)}
          </div>
        </div>
      </div>
    </section>
  )
}