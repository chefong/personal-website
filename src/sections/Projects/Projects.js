import styles from './Projects.module.scss';
import Card from './Card/Card';
import { FormattedMessage } from 'text-provider';
import { projects } from '../../shared/constants';

export default function Projects() {
  return (
    <section className={`${styles.section} container-fluid`} id="projects" data-testid="Projects-container">
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className={styles.headerGroup}>
            <h2 className={styles.header}>
              <span className={styles.headerNum}>02. </span>
              <FormattedMessage id="PROJECTS_HEADING" />
            </h2>
          </div>
        </div>
      </div>
      <div className="row justify-content-center">
        <div className="col-lg-9 col-md-9">
          <div className="row justify-content-center">
            {projects.map((project, index) => <Card { ...project } key={index} />)}
          </div>
        </div>
      </div>
    </section>
  );
}