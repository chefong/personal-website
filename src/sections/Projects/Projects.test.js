import React from 'react';
import { render } from '@testing-library/react';
import Projects from './Projects';
import { projects } from '../../shared/constants';

describe('Projects', () => {
  it('renders', () => {
    const { getByTestId } = render(<Projects />);
    expect(getByTestId('Projects-container')).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(<Projects />);
    expect(container).toMatchSnapshot();
  });

  it('contains correct number of project cards', () => {
    const { getAllByTestId } = render(<Projects />);
    expect(getAllByTestId('Card-container')).toHaveLength(projects.length);
  });
});
