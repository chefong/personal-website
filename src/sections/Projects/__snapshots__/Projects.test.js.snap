// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects matches snapshot 1`] = `
<div>
  <section
    class="undefined container-fluid"
    data-testid="Projects-container"
    id="projects"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-lg-9 col-md-9"
      >
        <div
          class="react-reveal"
        >
          <h2>
            <span>
              02. 
            </span>
            <span />
          </h2>
        </div>
      </div>
    </div>
    <div
      class="row justify-content-center"
    >
      <div
        class="col-lg-9 col-md-9"
      >
        <div
          class="row justify-content-center"
        >
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/willshiao/Patch"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
              <span
                class="badge badge-secondary"
              >
                <span />
              </span>
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/willshiao/CheaterBeater"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
              <span
                class="badge badge-secondary"
              >
                <span />
              </span>
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/KirbyDownB/GEICOChatBot"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
              <span
                class="badge badge-secondary"
              >
                <span />
              </span>
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/chefong/birdie"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/kimjihwan0208/stockoverflow"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
            </div>
          </div>
          <div
            class="col-lg-4 col-md-6"
          >
            <div
              data-testid="Card-container"
            >
              <a
                href="https://github.com/ericong18/joyride"
                target="_blank"
              />
              <p>
                <span />
              </p>
              <p>
                <span />
              </p>
              <span
                class="badge badge-secondary"
              >
                <span />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
